scs2-definition/src/main/java/us/ihmc/scs2/definition/robot/urdf/URDFTools.java

```diff
@@ -209,66 +214,188 @@ public static URDFModel loadURDFModel(InputStream inputStream, Collection<String
       {
-         try
+         mergedModel.setName(baseModel.getName());
+      }
+
+      // Merge links that haven't already been added
+      List<URDFLink> mergedLinks = new ArrayList<>();
+      if (baseModel.getLinks() != null)
+      {
+         mergedLinks.addAll(baseModel.getLinks());
+         for (URDFLink link : baseModel.getLinks())
          {
-            inputStream.close();
+            mergedLinkNames.add(link.getName());
          }
-         catch (IOException e)
+      }
+      if (additionalModel.getLinks() != null)
+      {
+         for (URDFLink link : additionalModel.getLinks())
          {
-            LogTools.error(e.getMessage());
+            if (!mergedLinkNames.contains(link.getName()))
+            {
+               mergedLinks.add(link);
+               mergedLinkNames.add(link.getName());
+            }
+            else
+            {
+               LogTools.warn("Link ({}) has already been added to urdf, skipping link", link.getName());
+            }
+         }
+      }
+      mergedModel.setLinks(new ArrayList<>(mergedLinks));
+
+      // Merge joints that haven't already been added
+      List<URDFJoint> mergedJoints = new ArrayList<>();
+      if (baseModel.getJoints() != null)
+      {
+         mergedJoints.addAll(baseModel.getJoints());
+         for (URDFJoint joint : baseModel.getJoints())
+         {
+            mergedJointNames.add(joint.getName());
+         }
+      }
+      if (additionalModel.getJoints() != null)
+      {
+         for (URDFJoint joint : additionalModel.getJoints())
+         {
+            if (!mergedJointNames.contains(joint.getName()))
+            {
+               mergedJoints.add(joint);
+               mergedJointNames.add(joint.getName());
+            }
+            else
+            {
+               LogTools.warn("Joint ({}) has already been added to urdf, skipping joint", joint.getName());
+            }
+         }
+      }
+      mergedModel.setJoints(new ArrayList<>(mergedJoints));
+
+      // Merge Gazebos that haven't already been added
+      List<URDFGazebo> mergedGazebos = new ArrayList<>();
+      if (baseModel.getGazebos() != null)
+      {
+         mergedGazebos.addAll(baseModel.getGazebos());
+         for (URDFGazebo gazebo : baseModel.getGazebos())
+         {
+            mergedGazeboNames.add(gazebo.getReference());
          }
       }
+      if (additionalModel.getGazebos() != null)
+      {
+         for (URDFGazebo gazebo : additionalModel.getGazebos())
+         {
+            if (!mergedGazeboNames.contains(gazebo.getReference()))
+            {
+               mergedGazebos.add(gazebo);
+               mergedGazeboNames.add(gazebo.getReference());
+            }
+            else
+            {
+               LogTools.warn("Gazebo ({}) has already been added to urdf, skipping gazebo", gazebo.getReference());
+            }
+         }
+      }
+      mergedModel.setGazebos(new ArrayList<>(mergedGazebos));
+
+      return mergedModel;
```

Comment:
The merged model contains all parts that are unique in both the baseModel and the additionalModel. By passing in the combinedURDFModel into this method at each loop iteration, we slowly build up a combined URDF